go/libraries/doltcore/env/environment.go

```diff
@@ -1267,15 +1268,22 @@ func (dEnv *DoltEnv) Unlock() error {
 }
 
 // WriteLockfile writes a lockfile encoding the pid of the calling process.
-func WriteLockfile(fs filesys.Filesys, lock DBLock) error {
+func WriteLockfile(fs filesys.Filesys, lock *DBLock) error {
+	// if the DoltDir doesn't exist, create it.
+	doltDir, _ := fs.Abs(dbfactory.DoltDir)
+	err := fs.MkDirs(doltDir)
```

Comment:
Creating the `.dolt` dir without initializing a db may cause problems for other commands... I believe other code looks for the presence of the `.dolt` dir and if found, assumes that it's a valid db and that other data files will be present in `.dolt`, so those commands may start erroring out in weird ways if that assumption doesn't hold. Definitely worth testing that ahead of time to see if it's going to create other problems.